### ISSUE TITLE
Authentication failed on Windows machine 

### ISSUE THREAD

**Issue by sabahirfan** [asterisk-java/asterisk-java#166]
I am writing a small application to monitor events and I am implementing ManagerEventListener interface.

This small program works fine on Mac. But when I run the same on Windows machine. It throws org.asteriskjava.manager.AuthenticationFailedException: Authentication failed

Your help will be highly appreciated to solve this issue. I have spent whole weekend struggling on this already.

Here is the code snippet from my small application:
```
public class PhoneNumberResolver implements ManagerEventListener {
private ManagerConnection managerConnection;

public void PhoneNumberResolver()  {
        ManagerConnectionFactory factory = new ManagerConnectionFactory(
                hostAddress, port, username, password);
        this.managerConnection = factory.createManagerConnection();
}

public void run() throws IOException, AuthenticationFailedException,
            TimeoutException, InterruptedException, IllegalStateException {

        managerConnection.addEventListener(this);
        managerConnection.login();

        while (true) ;
        // Thread.sleep(60000);
        // managerConnection.logoff();
    }

public void onManagerEvent(ManagerEvent event) {
logger.info("Event :" + event);
}
}
```

Here are the error logs:
----------------------------------------------------------------------------
2016-09-18 21:48:44.202  INFO 10300 --- [           main] o.a.m.internal.ManagerConnectionImpl     : Connecting to 54.217.225.96:35332
2016-09-18 21:48:44.374  INFO 10300 --- [tion-0-Reader-0] o.a.m.internal.ManagerConnectionImpl     : Connected via Asterisk Call Manager/1.3
2016-09-18 21:48:45.473  INFO 10300 --- [tion-0-Reader-0] **o.a.manager.internal.ManagerReaderImpl   : Terminating reader thread: No more lines available: null
2016-09-18 21:48:45.478  INFO 10300 --- [           main] o.a.m.internal.ManagerConnectionImpl     : Closing socket**.
2016-09-18 21:48:45.493 ERROR 10300 --- [           main] com.cds.ami.service.PhoneNumberResolver  : Authentication failed

**org.asteriskjava.manager.AuthenticationFailedException: Authentication failed**
        at org.asteriskjava.manager.internal.ManagerConnectionImpl.doLogin(ManagerConnectionImpl.java:639) ~[asterisk-java-1.1.0-beta.jar!/:na]
        at org.asteriskjava.manager.internal.ManagerConnectionImpl.login(ManagerConnectionImpl.java:499) ~[asterisk-java-1.1.0-beta.jar!/:na]
        at org.asteriskjava.manager.internal.ManagerConnectionImpl.login(ManagerConnectionImpl.java:484) ~[asterisk-java-1.1.0-beta.jar!/:na]
        at org.asteriskjava.manager.DefaultManagerConnection.login(DefaultManagerConnection.java:294) ~[asterisk-java-1.1.0-beta.jar!/:na]
        at com.cds.ami.service.PhoneNumberResolver.run(PhoneNumberResolver.java:63) ~[classes!/:na]
        at com.cds.ami.service.PhoneNumberResolver.start(PhoneNumberResolver.java:105) ~[classes!/:na]
        at com.cds.ami.AsteriskManagerInterfaceApplication.run(AsteriskManagerInterfaceApplication.java:29) [classes!/:na]
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:798) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:782) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at org.springframework.boot.SpringApplication.afterRefresh(SpringApplication.java:769) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:314) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1185) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1174) [spring-boot-1.4.0.RELEASE.jar!/:1.4.0.RELEASE]
        at com.cds.ami.AsteriskManagerInterfaceApplication.main(AsteriskManagerInterfaceApplication.java:39) [classes!/:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_45]
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_45]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:1.8.0_45]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[na:1.8.0_45]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [cds-sip-ami-0.0.1.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [cds-sip-ami-0.0.1.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [cds-sip-ami-0.0.1.jar:na]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:58) [cds-sip-ami-0.0.1.jar:na]

**Comment by sabahirfan** [comment]
I am using the below version of library. 

compile group: 'org.asteriskjava', name: 'asterisk-java', version: '1.0.0-final'

I also tried the latest version as well as different versions available. But the same error on Windows machine but works fine on Mac Book.

**Comment by rlsutton1** [comment]
I would check the permissions in manager.conf
I would also check the character encoding of the user name and password - I have seen many problems caused by the differences in character encoding between windows and mac

**Comment by sabahirfan** [comment]
Thanks a lot. I already did check the permissions. I will check the character encoding as well and will update you accordingly.

Many Thanks
Sabah

**Comment by sabahirfan** [comment]
Thanks a lot. I got it fixed.

It was neither permission issue nor the encoding issue.

I was using username key in my properties file. Strangely on Windows machine, it was picking up the operating system user name instead of the username defined in the properties file. I changed the key name from username to something else and it started working.

Many thanks for you pointers that actually helped me nailing down the root cause.

Cheers]Sabah
